source/material_model/visco_plastic.cc

```diff
@@ -354,13 +355,21 @@ namespace aspect
     evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
              MaterialModel::MaterialModelOutputs<dim> &out) const
     {
+      // set up additional output for the derivatives
+      MaterialModel::MaterialModelDerivatives<dim> *derivatives;
+      derivatives = out.template get_additional_output<MaterialModel::MaterialModelDerivatives<dim> >();
+
+      const double derivative_scaling_factor = &(this->get_simulator()) != nullptr ? (this->get_parameters().nonlinear_solver == Parameters<dim>::NonlinearSolver::Newton_Stokes ? this->get_newton_handler().get_newton_derivative_scaling_factor() : 0) : 1;
```

Comment:
If I remove the const I can spell it out in if-else statements. I could also add a comment explaining the line. Something like: If there is no simulator, it is a test environment and we want to enable the derivatives, so we set this variable to one. If we do not use the Newton solver, we do not need the derivatives, so we set this variable to zero. Otherwise, we use the value from the newton handler.

Reply:
If having the `const` leads to code that is hard to read, then get rid of it. The comment sounds good to me!